post-processor/vagrant/util.go

```diff
@@ -51,6 +51,23 @@ func CopyContents(dst, src string) error {
 	return nil
 }
 
+// Creates a (hard) link to a file, ensuring that all parent directories also exist.
+func LinkFile(dst, src string) error {
+	dstDir, _ := filepath.Split(dst)
+	if dstDir != "" {
+		err := os.MkdirAll(dstDir, os.ModePerm)
```

Comment:
I think we should use 0755 here

Reply:
Makes sense. I will also change the same line in the CopyContents function next to this one.